bls-keystore/src/main/java/tech/pegasys/signers/bls/keystore/model/SCryptParam.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.bls.keystore.model;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import tech.pegasys.signers.bls.keystore.KeyStoreValidationException;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import org.apache.tuweni.bytes.Bytes;
+import org.bouncycastle.crypto.generators.SCrypt;
+
+public class SCryptParam extends KdfParam {
+  private final int n;
+  private final int p;
+  private final int r;
+
+  /**
+   * SCrypt Key Derivation Function
+   *
+   * @param dklen The length of key to generate.
+   * @param n CPU/Memory cost parameter. Must be larger than 1, a power of 2 and less than <code>
+   *     2^(128 * r / 8)</code>.
+   * @param p Parallelization parameter. Must be a positive integer less than or equal to <code>
+   *     Integer.MAX_VALUE / (128 * r * 8)</code>
+   * @param r the block size, must be &gt;= 1.
+   * @param salt The salt to use
+   */
+  @JsonCreator
+  public SCryptParam(
+      @JsonProperty(value = "dklen", required = true) final int dklen,
+      @JsonProperty(value = "n", required = true) final int n,
+      @JsonProperty(value = "p", required = true) final int p,
+      @JsonProperty(value = "r", required = true) final int r,
+      @JsonProperty(value = "salt", required = true) final Bytes salt) {
+    super(dklen, salt);
+    this.n = n;
+    this.p = p;
+    this.r = r;
+  }
+
+  /**
```

Comment:
new?

Reply:
Yeah I copied this validator command branch. That has been merged now so this is the same as master